snuba/subscriptions/store.py

```diff
@@ -0,0 +1,104 @@
+import json
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Collection
+
+from rediscluster import StrictRedisCluster
+
+from snuba.subscriptions.data import Subscription
+from snuba.utils.codecs import Codec
+
+
+@dataclass(frozen=True)
+class RedisPayload:
+    __slots__ = ["key", "value"]
+
+    key: str
```

Comment:
Why do you need to add the key in the RedisPayload? That's always a string, and you do not use it to decode a value from the cache.
Also we have a cache api (like since yesterday) so we can follow that condition 
https://github.com/getsentry/snuba/blob/master/snuba/state/cache.py#L11-L18

Reply:
I was mostly following the pattern I saw in `CommitCodec` uses, but maybe it's unnecessary? It seemed helpful to be able to derive the key automatically.